### PR TITLE
Info about committing node_modules to the git repo

### PR DESCRIPTION
**Pull request by DaniAkash**
### Why:

Added some additional information about the problems the users might face if they commit their `node_modules` directory to git. Will help them take a more informed decision.

### What's being changed:

Added the following note:

---

**NOTE**

Checking in `node_modules` into a git repository is known to cause a [few problems](https://byjoeybaker.com/why-you-should-never-commit-node-modules-in-nodejs). Hence the following alternative approach exists

---

### Check off the following:
- [x] All of the tests are passing.
- [ ] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [ ] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [ ] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by janiceilene** [comment]
@DaniAkash Thanks so much for opening a PR! I'll get this triaged for review ⚡

**Comment by DaniAkash** [comment]
Will the message be added?

**Comment by janiceilene** [comment]
@DaniAkash We haven't gotten a chance to review this PR yet 💛 Thanks for your patience! Our small team is working our way through reviewing all of the amazing contributions ✨

**Comment by DaniAkash** [comment]
I had to add the comment cuz of github stale bot

File: content/actions/creating-actions/creating-a-javascript-action.md [hunk 1]
```diff
@@ -176,6 +176,14 @@ git tag -a -m "My first action release" v1
...
 
+---
+
+**NOTE**
+
+Checking in `node_modules` into a git repository is known to cause a [few problems](https://byjoeybaker.com/why-you-should-never-commit-node-modules-in-nodejs). Hence the following alternative approach exists
+
+---
+
 As an alternative to checking in your `node_modules` directory you can use a tool called [`@vercel/ncc`](https://github.com/vercel/ncc) to compile your code and modules into one file used for distribution.
```
Comment (lecoursen) [inline content/actions/creating-actions/creating-a-javascript-action.md:187]: Let's simplify this to reduce cognitive load on the reader. I'd be happy to include a link, following the [style guidelines](https://github.com/github/docs/blob/main/contributing/content-style-guide.md#links), if you can provide a more authoritative source. 

```suggestion
Checking in your `node_modules` directory can cause problems. As an alternative, you can use a tool called [`@vercel/ncc`](https://github.com/vercel/ncc) to compile your code and modules into one file used for distribution.
```

**Review by lecoursen — COMMENTED** [review]
Thanks @DaniAkash! I left one suggested change and will merge this PR after you apply it!

**Comment by DaniAkash** [comment]
@vercel/ncc has mentioned some clear [motivations](https://www.npmjs.com/package/@vercel/ncc#motivation) behind the project. So we can skip the link 👍

**Review by lecoursen — APPROVED** [review]

**Pull request merged**